testing/libneo4j-client/APKBUILD

```diff
@@ -21,17 +21,42 @@ build() {
             --prefix=/usr \
             || return 1
     make || return 1
-    make LIBS="-lfts" check || return 1
     make doc || return 1
 }
 
+check() {
+    cd "$_builddir"
+    make LIBS="-lfts" check || return 1
+}
+
 package() {
     cd "$_builddir"
     make DESTDIR="$pkgdir" install || return 1
     mkdir -p "$pkgdir"/usr/share/doc/libneo4j-client
     mv doc/html/* "$pkgdir"/usr/share/doc/libneo4j-client/
 }
 
+doc() {
+    depends="$depends_doc"
+    pkgdesc="$pkgdesc (documentation)"
+    install_if="docs $pkgname=$pkgver-r$pkgrel"
+
+    if [ "X$subpkgname" = "Xlibneo4j-client-doc" ]; then
+        mkdir -p "$subpkgdir"/usr/share
+        mv "$pkgdir"/usr/share/doc "$subpkgdir"/usr/share/
+    elif [ "X$subpkgname" = "Xneo4j-client-doc" ]; then
+        mkdir -p "$subpkgdir"/usr/share
+        mv "$pkgdir"/usr/share/man "$subpkgdir"/usr/share/
+        find "$subpkgdir"/usr/share/man/ -type f | xargs gzip -9
+    else
+        die "Unknown doc package $pkgname"
+    fi
+
+    # remove if empty, ignore error (not empty)
+    rmdir "$pkgdir"/usr/share "$pkgdir"/usr || true 2>/dev/null
+    return 0
+}
```

Comment:
This is all unnecessarily complicated. We usually don’t make multiple `-doc` subpackages from a single abuild. Why do you need it?

Reply:
It's building two different binary packages - a library and a shell client. The library has documentation file (HTML) and the shell has a man page. So I put them in separate doc packages.

Happy to do so differently if you can advise your preference.